src/main/java/org/spongepowered/api/asset/Asset.java

```diff
@@ -77,6 +78,24 @@ default void copyToFile(Path output) throws IOException {
     }
 
     /**
+     * Copes this Asset to the specified 'outputDirectory' {@Link Path}.
+     * @param outputDirectory directory to copy to
+     * @throws IOException
+     */
+    default void copyToDirectory(Path outputDirectory) throws IOException {
+        checkNotNull(outputDirectory, outputDirectory);
+        copyToFile(outputDirectory.resolve(this.getFileName()));
+    }
+
+    /**
+     *
+     * @return get's the last portion of the Asset URL, e.g. the file name
+     */
+    default String getFileName(){
+        return new File(getUrl().getPath()).getName();
```

Comment:
The original reasoning was the potential that an asset could contain file separators at the end of the Url, (I don't think it's applicable, but for instance WebAssets with query strings) But I think that would fail if I used a File or Path anyway, so I will split it, since it's in a default method if a strange implementation of Asset needs it it can override the functionality anyway.

I don't expect that standard assets will cause the bug to trigger.